### PR TITLE
@W-17064386 zero raised to a negative power is undefined

### PR DESCRIPTION
**Pull request by bairenlong-sfdc**

### CODE REVIEW

File: impl/src/main/java/com/force/formula/impl/FormulaSqlHooks.java [hunk 1]
```diff
@@ -812,6 +812,7 @@ default String sqlGreatest(String arg1, String arg2) {
     default SQLPair getPowerSql( String[] args, String[] guards) {
         String sql = "POWER(" + args[0] + ", " + args[1] + ")";
         String guard = SQLPair.generateGuard(guards, "TRUNC(" + args[1] + ")<>" + args[1] +
+                    " OR(" + args[0] + "=0 AND " + args[1] + "<0)" +
```
Comment (jfreschl) [inline impl/src/main/java/com/force/formula/impl/FormulaSqlHooks.java:815]: new guard looks good to me.

**Review by jfreschl — APPROVED** [review]

File: oracle-test/src/test/resources/com/force/formula/impl/formulaTestV2.xml [hunk 2]
```diff
@@ -27700,208 +27700,6 @@
 					Error: java.lang.ArithmeticException,
 					null"/>
    </testcase>
-   <testcase testName="testBigDivideWithFunc"
```
Comment (bairenlong-sfdc) [inline oracle-test/src/test/resources/com/force/formula/impl/formulaTestV2.xml:27703]: Maybe we should get rid if all of the oracle test cases in git, they were moved to core.

Comment (sahil-here) [reply oracle-test/src/test/resources/com/force/formula/impl/formulaTestV2.xml:27703]: @bairenlong-sfdc 
For Oracle tests, ideally we should keep them in git as the git formula-engine lib could be potentially used by any project apart from core as well and we should have coverage for that. If needed, we can remove these tests from Core, if only we think that the test's output will not change due to any specific Core hooks or Core Formula Engine Context(Api/Entity/Template/etc).

For SDB, we have to have the test in Core for sure as the SDB hooks are in Core.

Having said that, I think it is upto us to decide if it is easier to keep coverage in Core only and git might or might not have that coverage.

Comment (sahil-here) [reply oracle-test/src/test/resources/com/force/formula/impl/formulaTestV2.xml:27703]: The updates to the tests are not consistent here.

Since we are updating the logic for all SQL, it will impact all the substrates like Postgres, Oracle, Mysql, etc. Also, since we are adding a new scenario of negative powers, we need to add testData in the existing tests, that test the negative powers as well.

Therefore, all the following files will be impacted in Git:
1) Test Definition files for all database substrates. If you don't want to test other substrates apart from postgres and Oracle, you can remove the test defintions and corresponding gold files for those tests under that database substrates.
2) Gold Files for all database substrates.

Following files will be impacted in Core:
1) SDB test definition files to include new test data with negative powers.
2) SDB Gold Files for those tests.
3) Oracle test definitions files in Core to include new test data with negative powers. If you are coverning it in Git and don't want to cover it in Core, then you can remove that in Core or you can do vise-versa. But have the coverage atleast at one of the places.
4) Oracle Gold files in Core.
